### PR TITLE
add SPV_INTEL_fpga_argument_interfaces

### PR DESCRIPTION
**Pull request by broxigarchen**
Regarding PR for spec https://github.com/KhronosGroup/SPIRV-Registry/pull/177

added intel SPV_INTEL_fpga_argument_interfaces Enum

### CODE REVIEW

**Comment by broxigarchen** [comment]
A set of new enum are added in order to support intel sycl kernel argument annotations in spirv translator

**Comment by johnkslang** [comment]
This needs be done by updating the grammar and then using the tools to autogenerate the headers.

**Comment by broxigarchen** [comment]
> This needs be done by updating the grammar and then using the tools to autogenerate the headers.

done

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -14260,6 +14320,11 @@
           "extensions" : [ "SPV_INTEL_split_barrier" ],
           "version" : "None"
         },
+        {
+          "enumerant" : "FPGAArgumentInterfacesINTEL",
+          "value" : 6174,
+          "version" : "None"
```
Comment (bashbaug) [inline include/spirv/unified1/spirv.core.grammar.json:14326]: The new capability should be guarded by the name of the extension.  Please check that the extension name is correct below - it is SPV_INTEL_argument_interfaces in the PR title but I think the correct name of the extension is SPV_INTEL_fpga_argument_interfaces?

Note that only the capability should be guarded by the extension, and the new decorations should only be guarded by the new capability.

```suggestion
          "enumerant" : "FPGAArgumentInterfacesINTEL",
          "value" : 6174,
          "extensions" : [ "SPV_INTEL_fpga_argument_interfaces" ],          
          "version" : "None"
```

**Review by bashbaug — COMMENTED** [review]
Mostly LGTM but I found one issue.

Please be sure to remove "Draft" status when this is ready for broader working group review and merging - thanks!

**Review by bashbaug — COMMENTED** [review]
Thanks!  Latest version LGTM.  Needs a non-Intel reviewer before merging.

File: include/spirv/unified1/spirv.core.grammar.json [hunk 2]
```diff
@@ -11986,6 +11986,66 @@
...
+          "value" : 6176,
+          "capabilities" : [ "FPGAArgumentInterfacesINTEL" ],
+          "version" : "None"
+        },
+        {
+          "enumerant" : "MMHostInterfaceAddressWidthINTEL",
+          "value" : 6177,
+          "capabilities" : [ "FPGAArgumentInterfacesINTEL" ],
+          "version" : "None"
+        },
```
Comment (MrSidims) [inline include/spirv/unified1/spirv.core.grammar.json:12007]: It has a parameter as well as the others below.
```suggestion
          "enumerant" : "MMHostInterfaceAddressWidthINTEL",
          "value" : 6177,
          "parameters" : [
            { "kind" : "LiteralInteger", "name" : "'AddressWidth'" }
          ],
          "capabilities" : [ "FPGAArgumentInterfacesINTEL" ],
          "version" : "None"
        },
```

**Review by MrSidims — CHANGES_REQUESTED** [review]

File: include/spirv/unified1/spirv.core.grammar.json (hunk 2 above)
Comment (broxigarchen) [reply include/spirv/unified1/spirv.core.grammar.json]: done

**Review by MrSidims — APPROVED** [review]
LGTM

**Comment by bashbaug** [comment]
Here is the PR for the extension spec:
https://github.com/KhronosGroup/SPIRV-Registry/pull/177

Note that the name of the extension is correctly `SPV_INTEL_fpga_argument_interfaces` in the specification and in these changes and is not `SPV_INTEL_argument_interfaces` in the name of this PR (@broxigarchen might be best to fix the PR name to avoid confusion).

**Comment by broxigarchen** [comment]
> Here is the PR for the extension spec: [KhronosGroup/SPIRV-Registry#177](https://github.com/KhronosGroup/SPIRV-Registry/pull/177)
> 
> Note that the name of the extension is correctly `SPV_INTEL_fpga_argument_interfaces` in the specification and in these changes and is not `SPV_INTEL_argument_interfaces` in the name of this PR (@broxigarchen might be best to fix the PR name to avoid confusion).

Done. Thanks for pointing this out! I forgot to update the PR's name.

**Review by dneto0 — APPROVED** [review]
LGTM

**Pull request merged**